.github/workflows/fetch-ci-nightly-data.yaml

```
@@ -4,6 +4,15 @@ on:
   schedule:
     - cron: '0 4 * * *'
   workflow_dispatch:
+  on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'scripts/fetch-data-nightly.js'
+      # - 'scripts/fetch-data-pr.js'
+      # - '.github/workflows/fetch-ci-nightly-data.yaml'
```

Let's do this as discussed in the meeting. We can add the other script in the PR that introduces it.
```suggestion
      - 'scripts/fetch-data-nightly.js'
      - '.github/workflows/fetch-ci-nightly-data.yaml'
```